modules/etl/talend/talend-definition/src/main/java/com/liferay/talend/utils/UriUtils.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.talend.utils;
+
+import com.liferay.talend.runtime.apio.constants.ApioConstants;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import java.util.Collections;
+import java.util.Map;
+
+import javax.ws.rs.core.UriBuilder;
+
+import org.apache.commons.lang3.StringUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * @author Zoltán Takács
+ */
+public class UriUtils {
+
+	public static URI addQueryConditionToURL(
+		String resourceURL, String queryCondition) {
+
+		if (StringUtils.isEmpty(queryCondition)) {
+			if (_log.isDebugEnabled()) {
+				_log.debug("Query condition was empty!");
+			}
+
+			return getUri(resourceURL);
+		}
+
+		Collections.singletonMap(
+			ApioConstants.FILTER_QUERY_PARAM, queryCondition);
+
+		URI decoratedUri = updateWithQueryParameters(
```

Comment:
I'm thinking that, if the class is `URI`, the variable name should be `decoratedURI` (same for `URL`). What do you think?